src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -278,220 +263,229 @@ function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourceLoade
-            if (!assemblies) {
-              return BINDING.mono_obj_array_new(0);
-            }
-            const assemblyBytes = BINDING.mono_obj_array_new(assemblies.length);
-            for (let i = 0; i < assemblies.length; i++) {
-              const assembly = assemblies[i] as ArrayBuffer;
-              BINDING.mono_obj_array_set(assemblyBytes, i, BINDING.js_typed_array_to_array(new Uint8Array(assembly)));
-            }
-            return assemblyBytes;
-          };
-
-          Blazor._internal.readLazyPdbs = () => {
-            const { assemblies, pdbs } = lazyResources;
-            if (!assemblies) {
-              return BINDING.mono_obj_array_new(0);
-            }
-            const pdbBytes = BINDING.mono_obj_array_new(assemblies.length);
-            for (let i = 0; i < assemblies.length; i++) {
-              const pdb = pdbs && pdbs[i] ? new Uint8Array(pdbs[i] as ArrayBufferLike) : new Uint8Array();
-              BINDING.mono_obj_array_set(pdbBytes, i, BINDING.js_typed_array_to_array(pdb));
-            }
-            return pdbBytes;
-          };
-        }
+      if (resourceLoader.bootConfig.aspnetCoreBrowserTools) {
+        // See https://github.com/dotnet/aspnetcore/issues/37357#issuecomment-941237000
+        MONO.mono_wasm_setenv('__ASPNETCORE_BROWSER_TOOLS', resourceLoader.bootConfig.aspnetCoreBrowserTools);
+      }
 
-        return lazyResources['assemblies'].length;
-      }));
+      // -1 enables debugging with logging disabled. 0 disables debugging entirely.
+      MONO.mono_wasm_load_runtime(appBinDirName, hasDebuggingEnabled() ? -1 : 0);
+      MONO.mono_wasm_runtime_ready();
+      attachInteropInvoker();
+      onReady();
     };
-  });
-
-  module.postRun.push(() => {
-    if (resourceLoader.bootConfig.debugBuild && resourceLoader.bootConfig.cacheBootResources) {
-      resourceLoader.logToConsole();
-    }
-    resourceLoader.purgeUnusedCacheEntriesAsync(); // Don't await - it's fine to run in background
 
-    if (resourceLoader.bootConfig.icuDataMode === ICUDataMode.Sharded) {
-      MONO.mono_wasm_setenv('__BLAZOR_SHARDED_ICU', '1');
+    async function addResourceAsAssembly(dependency: LoadingResource, loadAsName: string) {
+      const runDependencyId = `blazor:${dependency.name}`;
+      Module.addRunDependency(runDependencyId);
 
-      if (resourceLoader.startOptions.applicationCulture) {
-        // If a culture is specified via start options use that to initialize the Emscripten \  .NET culture.
-        MONO.mono_wasm_setenv('LANG', `${resourceLoader.startOptions.applicationCulture}.UTF-8`);
+      try {
+        // Wait for the data to be loaded and verified
+        const dataBuffer = await dependency.response.then(r => r.arrayBuffer());
+
+        // Load it into the Mono runtime
+        const data = new Uint8Array(dataBuffer);
+        const heapAddress = Module._malloc(data.length);
+        const heapMemory = new Uint8Array(Module.HEAPU8.buffer, <any>heapAddress, data.length);
```

Comment:
```suggestion
        const heapMemory = new Uint8Array(Module.HEAPU8.buffer, heapAddress as any, data.length);
```

No big deal - it's just the style we use elsewhere.

Reply:
malloc now returns type `VoidPtr` which `new Uint8Array` doesn't know about.

Reply:
Will do.